src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -1027,75 +710,85 @@ public void resetCurrentStructure()
         currentStructure = null;
     }
 
+    /**
+     * Iterates through all entities and spawns them
+     * Suppressing Sonar Rule Squid:S3047
+     * The rule thinks we can merge the two forge loops iterating over resources
+     * But in this case the rule does not apply because that would destroy the logic.
+     */
+    @SuppressWarnings("squid:S3047")
     private Boolean spawnEntity(@NotNull final Structure.StructureBlock currentBlock)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)